### PR TITLE
feat(swap): suggest using TWAP

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Spec: https://docs.google.com/presentation/d/1ujFpnYbZ2eeaPA3v8DXRctWnDHrdlkx-63qScYrwlVk/edit#slide=id.g25b719a5370_0_55

The banner is displayed when:
1. Slippage is above 1%
2. AND the buy amount is greater than $50000

image

@fairlighteth I need your help with designing the banner 🙏

### CODE REVIEW

File: src/modules/swap/pure/banners/TwapSuggestionBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  if (!shouldSuggestTwap) return null
+
+  const routePath = parameterizeTradeRoute(tradeUrlParams, Routes.ADVANCED_ORDERS)
+
+  return (
+    <InlineBanner type="alert">
+      <p>
+        The price impact is {+priceImpact.toFixed(2)}%. Consider breaking up your order using{' '}
+        <StyledNavLink to={routePath}>TWAP orders</StyledNavLink>
```
Comment (alfetopito) [inline src/modules/swap/pure/banners/TwapSuggestionBanner.tsx:45]:
```suggestion
        a{' '}<StyledNavLink to={routePath}>TWAP order</StyledNavLink>
```

File: src/modules/swap/pure/banners/TwapSuggestionBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+`
+
+export interface TwapSuggestionBannerProps {
+  priceImpact: Percent | undefined
+  buyingFiatAmount: CurrencyAmount<Currency> | null
+  tradeUrlParams: TradeUrlParams
+}
+
+const PRICE_IMPACT_LIMIT = 1 // 1%
+const AMOUNT_LIMIT = 50_000 // $50,000
```
Comment (alfetopito) [inline src/modules/swap/pure/banners/TwapSuggestionBanner.tsx:29]: Use different amounts per network?

Comment (shoom3301) [reply src/modules/swap/pure/banners/TwapSuggestionBanner.tsx:29]: @elena-zh already asked about it, waiting a response from Anxo

**Comment by elena-zh** [comment]
Hey @shoom3301 , works great!
However, still there are pending [questions ](https://cowservices.slack.com/archives/C0361CDG8GP/p1690363842750309)on: 
1. Threshold amount for GC
2. Changing 'TWAP' to 'Advanced' in the link

And waiting for Styles from Michel :)

**Comment by shoom3301** [comment]
@elena-zh 
1. I set this config:
  [SupportedChainId.MAINNET]: 50_000, // $50,000
  [SupportedChainId.GNOSIS_CHAIN]: 5_000, // $5000
  [SupportedChainId.GOERLI]: 100, // $100
  
2. Changed, under your responsibility :)

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , can't check it for Mainnet and GC as I don't have enough funds there :)
However, changes for Goerli LGTM. 
Also, approving given the fact that @fairlighteth will enhance the banner soon :)

**Comment by elena-zh** [comment]
Hey @shoom3301 , could you please change a[ threshold for GC](https://cowservices.slack.com/archives/C0361CDG8GP/p1690451783345209?thread_ts=1690363842.750309&cid=C0361CDG8GP) to 500$?

**Review by alfetopito — APPROVED** [review]

**Comment by shoom3301** [comment]
@elena-zh done

**Review by elena-zh — APPROVED** [review]
LGTM! (still pending styles)

**Comment by shoom3301** [comment]
I merge it as it is, we can improve styles later

**Pull request merged**